pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,623 @@
+			return err
+		}
+	case DELETE:
+		exportFunction = deleteVirtualMachineExport
+		if err := handleDeleteFlags(args); err != nil {
+			return err
+		}
+	case DOWNLOAD:
+		exportFunction = downloadVirtualMachineExport
+		if err := handleDownloadFlags(args); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("Invalid function '%s'", funcName)
+	}
+
+	// VirtualMachineExport name
+	vmexportName = args[1]
+
+	return nil
+}
+
+// getVirtualMachineExport serves as a wrapper to get the VirtualMachineExport object
+func getVirtualMachineExport(client kubecli.KubevirtClient, namespace string) (*exportv1.VirtualMachineExport, error) {
+	vmexport, err := client.VirtualMachineExport(namespace).Get(context.TODO(), vmexportName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return vmexport, nil
+}
+
+// createVirtualMachineExport serves as a wrapper to create the virtualMachineExport object and, if needed, do error handling
+func createVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	vmexport, err := getVirtualMachineExport(client, namespace)
+	if err != nil {
+		return err
+	}
+	if vmexport != nil {
+		return fmt.Errorf("VirtualMachineExport '%s/%s' already exists", namespace, vmexportName)
+	}
+
+	vmexport = &exportv1.VirtualMachineExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmexportName,
+			Namespace: namespace,
+		},
+		Spec: exportv1.VirtualMachineExportSpec{
+			TokenSecretRef: getExportSecretName(),
+			Source:         exportSource,
+		},
+	}
+
+	vmexport, err = client.VirtualMachineExport(namespace).Create(context.TODO(), vmexport, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Generate/get secret to be used with the vmexport
+	_, err = getOrCreateTokenSecret(client, vmexport)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' created succesfully\n", namespace, vmexportName)
+	return nil
+}
+
+// deleteVirtualMachineExport serves as a wrapper to delete the virtualMachineExport object
+func deleteVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if err := client.VirtualMachineExport(namespace).Delete(context.TODO(), vmexportName, metav1.DeleteOptions{}); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		return fmt.Errorf("VirtualMachineExport '%s/%s' does not exist", namespace, vmexportName)
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' deleted succesfully\n", namespace, vmexportName)
+	return nil
+}
+
+// downloadVirtualMachineExport handles the process of downloading the requested volume from a VirtualMachineExport object
+func downloadVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if shouldCreate {
+		if err := createVirtualMachineExport(client, namespace); err != nil {
+			return err
+		}
+	}
+
+	defer deleteVirtualMachineExport(client, namespace)
```

Comment:
do we want to delete the vmexport after every download? even if we didnt create it in the virtctl command?

Reply:
The idea is that both create and download should be used along and that it'd be better to delete all virtctl traces in the cluster. I'm not sure if we should delete the object if it's not been created with virtctl, since that's not the expected use case. I guess that no vmexport should be needed after downloading the volume. I ping @awels since we discussed this some time ago.